src/nupic/algorithms/Connections.hpp

```diff
@@ -51,9 +51,9 @@ namespace nupic
       typedef Real32 Permanence;
       typedef UInt64 Iteration;
 
-      #define CELL_MAX (USHRT_MAX-1)
-      #define SEGMENT_MAX (UCHAR_MAX-1)
-      #define SYNAPSE_MAX (UCHAR_MAX-1)
+      // Defaults
+      static const unsigned char MAX_SEGMENTS_PER_CELL = 255;
```

Comment:
I wanted to keep the type the same as the variable that is bounding.  That way you have to change the type of both variables together if you want to increase the bound to something larger. Do you disagree?